ci/do_ci.sh

```diff
@@ -112,30 +112,30 @@ EOF
   examples/plugin/load/load_plugin_example ${PLUGIN_DIR}/libexample_plugin.so /dev/null
   exit 0
 elif [[ "$1" == "bazel.test" ]]; then
-  bazel build $BAZEL_OPTIONS //...
-  bazel test $BAZEL_TEST_OPTIONS //...
+  bazel build $BAZEL_OPTIONS -- //... -//exporters/etw/...
```

Comment:
@jsuereth - Josh, we need your help on this one. While manually excluding works, I don't think this is the proper way of doing it. Is there a better option to exclude some piece of functionality from Bazel build for POSIX, or - only include if building on Windows? something like `#if WIN32` or `if WIN32` ?

Reply:
I found this way to skip incompatible targets: [https://docs.bazel.build/versions/master/platforms.html#skipping-incompatible-targets](link)

But the CI gives an error saying ` no such attribute 'target_compatible_with' in 'cc_library' rule`, need to investigate properly.

Reply:
seems like, it's not yet part of the official release -- hence the error

Reply:
Is it possible to exclude running on non-pox environment in the bazel build file under //exporters/etw/...? In case the bazel build is invoked via the `do_ci.sh` script.

Reply:
@ThomsonTan - I think we discussed this issue, how to exclude - and appears like we converge on using "-" for now. It's been discussed in the comment above: https://github.com/open-telemetry/opentelemetry-cpp/pull/376#discussion_r523107518

Reply:
Still waiting for 3.8 release of BAZEL before we do anything fancy here.  The pending release has a decent experience to limit building non-compatible things (the `target_compatible_with` feature), otherwise we'll have wonky build flags all over.

TL;DR: for Bazel
- If you're adding new TARGETs (e.g. an exporter `cc_library`), the next major bazel release makes this easy.
- If you're adding different SOURCE on existing target (`cc_library` that has different `-D` or source files per platofrm) we can do this (easily) today.


I suggest we do one of the following:

1. We don't write build rules for ETW in bazel right now (unless you need them).   Open a ticket to add BAZEL support for the ETW exporter and we can block it on the 3.8 release of bazel.  This has the *least* impact for folks like myself who use bazel for development and want `bazel test //...` to work.
2. We use bazel's switch statement to make the ETW export compile a no-op plugin on non-windows platforms.  (i.e. it selects a source file that has nothing in it.  We can open a bug to rework this when bazel 3.8 hits.
3. You do what you've done here.  I think this impacts bazel users dramatically because `test //...` is broken on non-windows environments.  I'd prefer either (1) or (2).

Reply:
Can we just do `#ifdef _WIN32` , in all relevant ETW files. And unconditionally build. Then the `.cc` files are gonna produce 0-byte objects, that could safely be ignored?

Reply:
wrapped with `ifdef _WIN32` and removed all the wonky workarounds in `ci/do_ci.sh`, it's all clean now. thanks, @jsuereth @maxgolov for your inputs on this!

Reply:
Yeah, would be good to open a ticket to clean this up in a follow on PR.